workflow/controller/operator.go

```diff
@@ -1098,7 +1098,7 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 			newDaemonStatus = pointer.BoolPtr(true)
 			woc.log.Infof("Processing ready daemon pod: %v", pod.ObjectMeta.SelfLink)
 		}
-		woc.cleanUpPod(pod)
+		woc.queuePodForCleanup(pod, tmpl)
```

Comment:
can you leave this func's name uncnhanged, we'll want to back-port to v3.0 and this could create merge conflicts

Reply:
Done. Although the additional arg `tmpl` is needed to work with `tmpl.IsMainContainerName` in case `ContainerSet` is used. Would that be okay?